tasks/Common.cpp

```diff
@@ -229,20 +234,26 @@ void Common::pushFrame(GstElement* element, GstVideoInfo& info, Frame const& fra
     /* Create a buffer to wrap the last received image */
     GstBuffer* buffer = gst_buffer_new_and_alloc(info.size);
```

Comment:
I'm pretty sure this is wrong in the compressed case. There is no way GstVideoInfo can know the size of the compressed frame.

Clearly, you also have to separate the raw and compressed cases into two methods.